### DISCUSSION TITLE
How do get Pair Extraordinaire badge?

### DISCUSSION THREAD

**Discussion by Swift-Bunny** [community/community#20701] [Other Feature Feedback, Questions, & Ideas] [answered]
<img width="148" alt="pair-extraordinaire-default" src="https://user-images.githubusercontent.com/74031505/179380634-7ba599c0-d51d-410a-a1d8-c9706e2d80fd.png">
https://github.com/Schweinepriester/github-profile-achievements

**Comment by agetbase** [comment, marked as answer]
you need to create co-authored commits on the command line(in a terminal way).

e.g.
![스크린샷 2022-07-17 오전 11 08 49](https://user-images.githubusercontent.com/56868605/179381004-0de74dba-1464-48bd-9172-6811ec27a01d.png)

One thing to note is that the closing quotation mark(") must be written
after the co-author's email, and the email must be enclosed in <>

(more details )
https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors

**Reply by HwangTaehyun** [reply]
Thanks!

**Reply by ayyzenn** [reply]
In **name** should we type the actual name of the person or the username? 
Plus how to use **no-reply** for email to protect?

**Reply by dvdgitman** [reply]
If a person chooses to keep their email address private, you should use their GitHub-provided no-reply email to protect their privacy. Otherwise, the co-author's email will be available to the public in the commit message. If you want to keep your email private, you can choose to use a GitHub-provided no-reply email for Git operations and ask other co-authors to list your no-reply email in commit trailers.
and FYI I think you need also to merge after the commit., and yes you use the username in `name`

**Reply by ayyzenn** [reply]
Thanks!!

**Reply by Paulracisz** [reply]
This didn't work, maybe it takes multiple days to update, or it cannot be a repository that I own?

**Reply by eldare** [reply]
@Paulracisz did it work?

**Reply by agetbase** [reply]
oh, I'm sorry for the late response ;(

It still works now. But there are a few things to follow.
First of all, you can do it in your `own repository` or `others`.

**1. Create a new branch and commit changes to the branch.**
**2. Collaborating with team members.**
Create co-authored commits on the command line(in a terminal way).

# e.g.
> [!warning]
> The most important thing is not to write the co-authored information(`username`, `email`) in the next line immediately after > the commit message, but to enter it after at least two or three lines as below.

## Failed case
> [!caution]
> ![2](https://github.com/community/community/assets/56868605/360c3ad4-159d-4533-87da-24f63eeeb1a4)

## Success case
> [!note]
> Case(1)
> ![1](https://github.com/community/community/assets/56868605/b142daf3-463d-4a41-b677-a169c392c27d)
>
> Case(2)
> ![0](https://github.com/community/community/assets/56868605/c4a557e1-2a4f-4e78-b3be-4f963fcfaa85)

**3. Creating a Pull Request(`PR`)**
**4. Merging with `default Merge`(Create a merge commit) or `Squash and Merge`**

If you follow the above process, it will definitely work.
If there is a problem, I'll check it again. thx xD

**Comment by SUBHADIPMAITI-DEV** [comment]
1. create a branch and commit changes
2. get a collaborator to commit changes on the same branch
3. create a pull request
4. merge to the default branch using squash and merge

**Comment by Sriansh-raj** [comment]
Here is the public repo on list of available **GitHub achievements** and how to **get them**

Repo: **https://github.com/Sriansh-raj/GitHub-Achievements**

You can star it for your reference 🌟

Happy GitHub :octocat:

❤️

**Comment by Prakash4844** [comment]
Easiest way is to just use this repo that automates the process: https://github.com/Prakash4844/Get-Pair-Extraordinaire

**Reply by queenofcorgis** [reply]
GitHub's [Acceptable Use Policies](https://docs.github.com/en/site-policy/acceptable-use-policies/github-acceptable-use-policies) prohibits coordinated or inauthentic activity. As a result, we'll be hiding your comments. Please note any future violations may result in a temporary or indefinite block from the Community. Thanks for understanding.

**Comment by BhagyaAmarasinghe** [comment]
Pair extraordinaire can be achieved by co-authoring a commit. You have to include `Co-authored-by: NAME <NAME@EXAMPLE.COM>` in your commit message. For exmple:

```
$ git commit -m "Refactor usability tests.
>
>
Co-authored-by: NAME <NAME@EXAMPLE.COM>
Co-authored-by: ANOTHER-NAME <ANOTHER-NAME@EXAMPLE.COM>"
```
Github documentation for co-authoring commits:
https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors

**Reply by JeffreyYAJ** [reply]
Choose a repository where you have write access (your own or one where you are a collaborator).
Create a New Branch
Do not work directly on the main branch. Create a new feature branch (e.g., git checkout -b add-feature).
Co-author the Commit
You must include your collaborato's information in the commit message 
Type your message, leave a blank line, and add the co-author trailer exactly as shown below:
```bash
git commit -m "Update README documentation

Co-authored-by: Name <username@users.noreply.github.com>"
```

use this for reference

https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors

**Comment by yogithesymbian** [comment]
thankyou bro 🤝

**Reply by Octorcat** [reply]
great!

**Reply by Py-Ec** [reply]
Thank you

**Comment by shant00** [comment]
Github documentation for co-authoring commits:
https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors

its work fine

**Comment by t0linixer** [comment]
I tried this 
![image](https://github.com/community/community/assets/167388906/46911311-df09-489e-8dd3-88c91f488823)
but it doesnt works

**Reply by Prakash4844** [reply]
Try this repo https://github.com/Prakash4844/Get-Pair-Extraordinaire
